Avoid reading state from register on startup

DaveCTurner opened the pull request:
Now that #94497 ensures we read the state from the register during the election-winning process, there's no need to read it at startup. With this commit we start with an empty state every time.

DaveCTurner committed:
Avoid reading state from register on startup

Now that #94497 ensures we read the state from the register during the
election-winning process, there's no need to read it at startup. With
this commit we start with an empty state every time.

DaveCTurner commented:
This seems plausible but unfortunately doesn't quite work yet: I get (very) occasional failures in stabilisation even with the longer timeout. I will dig into them more deeply tomorrow.

DaveCTurner committed:
Merge branch 'main' into 2023-03-23-update-max-term-seen-on-election-failure

DaveCTurner committed:
Add simple test

DaveCTurner commented:
At least one of the failures was (largely) unrelated and I expect https://github.com/elastic/elasticsearch/pull/94726 will fix it.

fcofdez commented on server/src/test/java/org/elasticsearch/cluster/coordination/AtomicRegisterCoordinatorTests.java:
```diff
@@ -183,6 +183,17 @@ public void testCannotJoinClusterWithDifferentUUID() {
+
+    @Override
+    protected long extraStabilisationTime() {
+        return 300000; // TODO liveness improvements
```
Is this still needed with the fix from https://github.com/elastic/elasticsearch/pull/94726?

DaveCTurner replied:
Possibly not. I expect we will  need to allow for _some_ extra time for stabilisation in these tests, especially if we make some of the possibly-async steps into genuinely-async ones. But this wasn't that, it's just the first thing I try when looking at these tests.

DaveCTurner committed:
Merge branch 'main' into 2023-03-23-update-max-term-seen-on-election-failure

DaveCTurner committed:
WIP

DaveCTurner committed:
TODO awaitsfix broken test

DaveCTurner committed:
Longer timeouts

DaveCTurner committed:
Disable lag detector in testAckListenerReceivesNoAckFromHangingFollower

This test implicitly assumes that the final `stablise()` phase will end
before the hanging follower is removed from the cluster by the lag
detector, but changes elsewhere in the system may invalidate this
assumption. This commit substantially extends the lag detection timeout
to avoid this problem.

DaveCTurner committed:
Reinstate test

DaveCTurner committed:
Reverts

DaveCTurner committed:
#94758 not necessary to fix here any more

DaveCTurner committed:
Merge branch 'main' into 2023-03-23-update-max-term-seen-on-election-failure

fcofdez reviewed (approved):
LGTM, nice change :)
